mock-rollout/src/store/HardwareContext.tsx

```diff
@@ -2,37 +2,41 @@ import React, { createContext, FC, useEffect, useState, useReducer } from 'react
 import { IHardware } from '../model/HardwareModel';
 import { HardwareService } from '../services/HardwareService';
 
-interface IState {
-    hardwares?: IHardware[] | null | undefined;
+interface IHardwareContextState {
+    hardwares?: IHardware[];
     error?: string;
+}
+
+interface IHardwareContext {
+    hardwareState: IHardwareContextState,
+    dispatchHardwareState?: (action: IAction) => void,
     updateHardware?: (id: number, hardware: IHardware) => void
```

Comment:
this is a method to access the API, follow this example and add the getHardwareMethod which accesses the API same as this one